### PR TITLE
[3.0] Fix validation exception

### PR DESCRIPTION
**Pull request by AlbumenJ**
## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Review by juzi214032 — COMMENTED** [review]
Are there any problems with the current validator?

**Comment by AlbumenJ** [comment]
> Are there any problems with the current validator?

See: https://github.com/apache/dubbo-samples/blob/master/dubbo-samples-validation/src/test/java/org/apache/dubbo/samples/validation/ValidationServiceIT.java

File: dubbo-filter/dubbo-filter-validation/src/main/java/org/apache/dubbo/validation/support/jvalidation/JValidatorNew.java [hunk 1]
```diff
@@ -293,7 +292,7 @@ public void validate(String methodName, Class<?>[] parameterTypes, Object[] argu
             }
         } catch (ValidationException e) {
             // only use exception's message to avoid potential serialization issue
-            throw new RpcException(RpcException.VALIDATION_EXCEPTION, e.getMessage(), e);
+            throw new ValidationException(e.getMessage());
```
Comment (CrazyHZM) [inline dubbo-filter/dubbo-filter-validation/src/main/java/org/apache/dubbo/validation/support/jvalidation/JValidatorNew.java:295]: I'm a little concerned about affecting users who rely on `RpcException.VALIDATION_EXCEPTION`

Comment (AlbumenJ) [reply dubbo-filter/dubbo-filter-validation/src/main/java/org/apache/dubbo/validation/support/jvalidation/JValidatorNew.java:295]: RpcException.VALIDATION_EXCEPTION is bought by https://github.com/apache/dubbo/pull/9552/files#diff-d328eb1d6b3db5068164007286dffcd60b8419b3bfa0d7ebc566635f39f5b60fR42, which has not been published

**Pull request merged**